dali/test/python/test_operator_reduce.py

```diff
@@ -0,0 +1,113 @@
+import nvidia.dali.fn as fn
+import nvidia.dali.types as types
+import nvidia.dali.ops as ops
+
+from nvidia.dali.pipeline import Pipeline
+import numpy as np
+
+
+class Batch1D:
+    def __init__(self, data_type):
+        self._data_type = data_type
+
+    def __call__(self):
+        return [
+            np.array([ 1, 2,  3,  4], dtype = self._data_type),
+            np.array([99, 2, 10, 10], dtype = self._data_type)]
+
+    def valid_axes(self):
+        return [None, 0]
+
+
+class Batch2D:
+    def __init__(self, data_type):
+        self._data_type = data_type
+
+    def __call__(self):
+        return [
+            np.array([[ 1, 0, 2], [ 3, 1, 4]], dtype = self._data_type),
+            np.array([[ 5, 0, 6], [ 7, 0, 8]], dtype = self._data_type),
+            np.array([[ 1, 0, 2], [ 3, 1, 4]], dtype = self._data_type),
+            np.array([[ 5, 0, 6], [ 7, 0, 8]], dtype = self._data_type)]
+
+    def valid_axes(self):
+        return [None, 0, 1, (0, 1)]
+
+
+class Batch3D:
+    def __init__(self, data_type):
+        self._data_type = data_type
+
+    def __call__(self):
+        return [
+            np.array([[[1, 0, 1], [2, 3, 1]], [[0, 4, 1], [0, 4, 1]]], dtype = self._data_type),
+            np.array([[[5, 0, 1], [6, 7, 1]], [[0, 8, 1], [0, 4, 1]]], dtype = self._data_type)]
+
+    def valid_axes(self):
+        return [None, 0, 1, 2, (0, 1), (0, 2), (1, 2), (0, 1, 2)]
+
+
+def run_dali(reduce_fn, batch_fn, keep_dims, axes):
+    batch = batch_fn()
+    batch_size = len(batch)
+
+    # Needed due to how ExternalSource API works. It fails on methods, partials.
+    def get_batch():
+        return batch
+
+    pipe = Pipeline(batch_size=batch_size, num_threads=4, device_id=0)
+
+    with pipe:
+        input = fn.external_source(source = get_batch)
```

Comment:
Can't you just:
```suggestion
        input = fn.external_source(source = batch_fn)
```

Reply:
No. `ExternalSource` API for now does not work with some callables. I tried method and partial and it failed no a check inside. I think it could be reworked but not in this PR. For now I just wrapped it with ad hoc regular function so it works. 

I put the comment about it.